src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -2312,7 +2320,9 @@ cdef class Polynomial(CommutativePolynomial):
                         else:
                             return (self//h).any_root(ring, -degree, True)
         else:
-            return self.roots(ring=ring, multiplicities=False)[0]
+            if (rs := self.roots(ring=ring, multiplicities=False)):
+                return rs[0]
+            raise ValueError("no roots G %s" % self)
```

Comment:
```suggestion
            rs = self.roots(ring=ring, multiplicities=False)
            if rs:
                return rs[0]
            raise ValueError("no roots G %s" % self)
```
Creating one extra reference should not have any noticeable impact on the speed, but it makes the code easier to read.

I don't like the error message format (that extends to the entire method), but I would say this doesn't fit into the previous block as it is specifically for non-fields. I would instead do `"no roots (non field) %s" % self`.